terraform/lib/dependabot/terraform/file_fetcher.rb

```diff
@@ -37,22 +38,45 @@ def fetch_files
 
       private
 
+      sig { returns(T::Array[Dependabot::DependencyFile]) }
       def terraform_files
-        @terraform_files ||=
+        @terraform_files ||= T.let(
           repo_contents(raise_errors: false)
           .select { |f| f.type == "file" && f.name.end_with?(".tf") }
-          .map { |f| fetch_file_from_host(f.name) }
+          .map { |f| fetch_file_from_host(f.name) },
+          T.nilable(T::Array[Dependabot::DependencyFile])
+        )
       end
 
+      sig { returns(T::Array[Dependabot::DependencyFile]) }
       def terragrunt_files
-        @terragrunt_files ||=
+        @terragrunt_files ||= T.let(
           repo_contents(raise_errors: false)
           .select { |f| f.type == "file" && terragrunt_file?(f.name) }
-          .map { |f| fetch_file_from_host(f.name) }
+          .map { |f| fetch_file_from_host(f.name) },
+          T.nilable(T::Array[Dependabot::DependencyFile])
+        )
+      end
+
+      sig { params(file_name: String).returns(T::Boolean) }
+      def terragrunt_file?(file_name)
+        !lockfile?(file_name) && file_name.end_with?(".hcl")
       end
 
+      sig { params(filename: String).returns(T::Boolean) }
+      def lockfile?(filename)
+        filename == ".terraform.lock.hcl"
+      end
```

Comment:
These are the 2 methods I copied from `FileSelector`.

Reply:
It'd be nice to keep one implementation of these methods, because I can definitely see the logic around identifying if a file is a terragrunt file or lockfile change over time, and now we'd have to track down all the implementations. Probably the easiest thing to do here is to extract a new module for these and include it in both places that need it.

Reply:
I moved them to a `FileFilter` module